### PR TITLE
sensors:baro: add lsp25xx barometer driver

### PR DESCRIPTION
**Pull request by niewim19**
JIRA: PP-6

## Description

Adding lsp25xx barometer driver to sensor manager.

## Motivation and Context

Due to hardware issues this was not tested on the target, thus only draft PR is issued.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: sensors/baro/lps25xx.c [hunk 1]
```diff
@@ -16,61 +16,151 @@
...
 
 
-static inline time_t lps25xx_timeMsGet(void)
+static uint32_t translatePress(uint8_t lbyte, uint8_t mbyte, uint8_t hbyte)
 {
-	time_t now;
-	gettime(&now, NULL);
+	uint32_t val = 0;
+	val |= ((uint32_t)lbyte) | ((uint32_t)mbyte << 8) | ((uint32_t)hbyte << 16);
+	val &= 0x7FFFFF;                  /* dismiss highest bit as it is unusable */
```
Comment (MaciejPurski) [inline sensors/baro/lps25xx.c:72]: More readable version, a compiler would know, what conversion should be applied here:
```suggestion
	uint32_t val;

	val = lbyte | (mbyte << 8) | (hbyte << 16);
	/* dismiss highest bit as it is unusable */
	val &= 0x7FFFFF;
```

File: sensors/baro/lps25xx.c [hunk 2]
```diff
@@ -16,61 +16,151 @@
...
+	return val * SENSOR_CONV_PASCALS; /* sensor -> Pa */
+}
+
 
-	return now / 1000;
+static uint32_t translateTemp(uint8_t hbyte, uint8_t lbyte)
+{
+	uint32_t val = 0;
+	val |= (uint32_t)lbyte | ((uint32_t)hbyte << 8);
+	return val * SENSOR_CONV_KELV + SENSOR_KELV_OFFS;
```
Comment (MaciejPurski) [inline sensors/baro/lps25xx.c:81]:
```suggestion
	uint32_t val;

	val = lbyte | (hbyte << 8);

	return val * SENSOR_CONV_KELV + SENSOR_KELV_OFFS;
```

File: sensors/baro/lps25xx.c [hunk 3]
```diff
@@ -16,61 +16,151 @@
...
 }
 
 
-static void lps25xx_publishthr(void *data)
+static int spiWriteReg(spimsg_ctx_t *spiCtx, uint8_t regAddr, uint8_t regVal)
 {
-	sensor_info_t *info = (sensor_info_t *)data;
-	lps25xx_ctx_t *ctx = info->ctx;
+	unsigned char cmd[2] = { (regAddr & 0x7F), regVal }; /* write bit set to regAddr */
```
Comment (MaciejPurski) [inline sensors/baro/lps25xx.c:87]: As in the other PR, I don't know if this comment is necessary.

File: sensors/baro/lps25xx.c [hunk 4]
```diff
@@ -16,61 +16,151 @@
...
 
-	if (ctx == NULL) {
-		return -ENOMEM;
+	while (1) {
+		/* ODR set to 25Hz */
+		usleep(40 * 000);
+
+		obuf = REG_DATA_OUT | SPI_READ_BIT | SPI_AUTOADDRINCR_BIT;
+		err = spimsg_xfer(spiCtx, &obuf, 1, ibuf, SENSOR_OUTPUT_SIZE, 1);
+		gettime(&tstamp, NULL);
```
Comment (MaciejPurski) [inline sensors/baro/lps25xx.c:144]: As in the previous PR, `gettime()` might be done only in case of a success.

**Review by MaciejPurski — DISMISSED** [review]
Very small remarks, not a merge blocker to me.

**Review by MaciejPurski — DISMISSED** [review]
Looks good to me.

**Comment by niewim19** [comment]
Last changes made after testing code on hardware and finding that output temperature value is wrongly calculated

**Review by MaciejPurski — APPROVED** [review]

**Pull request merged**